sql/moz-fx-data-shared-prod/telemetry_derived/ech_adoption_rate_v1/query.sql

```diff
@@ -0,0 +1,150 @@
+WITH unioned AS (
```

Comment:
Do we really need multiple queries and `UNION ALL` here? IIRC GLAM does something similar without an union, but I might me wrong.

Reply:
Each of these queries is basically a short version of the ETL, which is much simpler for one or a few metrics, especially for metrics of different types. The full GLAM ETL here would be impractical, because there are too many steps there that don't make much sense for a small amount of metrics. That said, I think I can aggregate all of them into one statement, which will lose readability with maybe better performance, but this already runs in less than 3 minutes...